apps/resources/models.py

```diff
@@ -115,3 +116,29 @@ def get_most_tried(user=None, limit=1):
             '-most_tried',
         )[:limit]
         return resources
+
+    def get_related(self, user=None, limit=3):
+        data = {}
+        results = {}
+        ids_with_tags = Resource.objects.approved(
+            user=user,
+        ).exclude(
+            id=self.id,
+        ).values(
+            'id',
+            'tags',
+        )
+        for item in ids_with_tags:
```

Comment:
...because then you could `for resource_id, tags in id_with_tags` - bit ugly dict code.